zzk/service/hostregistry.go

```diff
@@ -77,11 +78,22 @@ type HostRegistryListener struct {
 
 // InitHostRegistry initializes the host registry
 func InitHostRegistry(conn client.Connection) error {
-	err := conn.CreateDir(hostregpath())
-	if err == client.ErrNodeExists {
-		return nil
+	for {
+		done := make(chan error, 1)
+		go func() {
+			done <- conn.CreateDir(hostregpath())
+		}()
+		select {
+		case err := <-done:
+			if err == client.ErrNodeExists {
+				return nil
+			}
+			return err
+		case <-time.After(5 * time.Second):
```

Comment:
What does it mean to retry forever?  what does that do to  zookeeper to keep callking createdir if it isn't returning?  Is there a case where it doesn't return?